### DISCUSSION TITLE
Installing Ipopt and ThirdParty on MacOS - Metis issue?

### DISCUSSION THREAD

**Discussion by kwmcbride** [coin-or/Ipopt#471] [Q&A] [answered]
Hi,

First I want to say thanks for the work done on maintaining the coin-or library of solvers. They are critical to my everyday work. Ipopt is perhaps the most important of these.

I have a new mac book with the M1 chip. I am trying to install Ipopt with the HSL libraries. I cannot get the HSL or Mumps to install due to a missing metis library. For example:

##################################################
##################################################

ld: library not found for -lmetis
collect2: error: ld returned 1 exit status
make[1]: *** [libcoinmumps.la] Error 1
make: *** [all] Error 2

Build failed, see error output above

I have install metis using brew and have confirmed that it is on PATH. I am not an expert in compiling C or C++, so I am not sure how to troubleshoot this. 

I should say that I am new to MacOS and have no issues installing solvers on my Linux computer.

I would appreciate any tips you may be able to provide. If I cannot fix this, the computer goes back to Apple!

Thanks

**Comment by svigerske** [comment, marked as answer]
That looks as if configure decided that the Metis lib is present, but when linking the Mumps library, it is not found.
Maybe that can happen if you use clang for the C/C++ part, but GCC for the Fortran part, because the Metis check in configure is done with the C compiler, but linking the lib is done with the Fortran compiler.
If that is the case, you want to make sure that GCC is used for C/C++ sources, too. For that you would pass something like `CC=gcc CXX=g++` to configure (or coinbrew). On the [travis builds](https://github.com/coin-or/Ipopt/blob/stable/3.14/.travis.yml), I also need `--with-metis-cflags=-I/usr/local/Cellar/metis/5.1.0/include --with-metis-lflags="-L/usr/local/Cellar/metis/5.1.0/lib -lmetis"`

**Reply by kwmcbride** [reply]
Thanks for the quick response. I can build HSL and Mumps by passing all of the arguments you provided. Simply using the CC and CXX arguments did not solve the issue.

To get the Ipopt build to work I had to use CC=gcc-11 and CXX=g++-11 due to aliasing with clang with gcc and g++. 

So for anyone else who may be reading this, the following is what worked for me:
```
coinbrew build Ipopt CC=gcc-11 CXX=g++-11 --with-metis-cflags=-I/usr/local/Cellar/metis/5.1.0/include --with-metis-lflags="-L/usr/local/Cellar/metis/5.1.0/lib -lmetis"
````
Again, thanks a lot for your help!
